File: MedArrival-Front/src/models/ArrivalDTO.ts
import { BaseDTO } from "./BaseDTO";
import { SaleDTO } from "./SaleDTO";
import { SupplierDTO } from "./SupplierDTO";

export interface ArrivalDTO extends BaseDTO {
    arrivalDate: Date;
    invoiceNumber: string;
    sales: SaleDTO[];
    supplier: SupplierDTO;
}

File: MedArrival-Backend/src/main/java/com/cnesten/medarrivalbackend/DTO/ArrivalDTO.java
package com.cnesten.medarrivalbackend.DTO;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.Set;

@Data
public class ArrivalDTO {
    private Long id;
    private LocalDateTime arrivalDate;
    private String invoiceNumber;
    private Set<SaleDTO> sales;
    private SupplierDTO supplier;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
